### DISCUSSION TITLE
USB DAC vs I2S DAC

### DISCUSSION THREAD

**Discussion by apastuszak** [rern/rAudio#423] [General]
I was using raudio1 originally with a HiFiBerry DAC2 HD and had very few problems. Then I added a touchscreen, which required a new case. The new case would not fit the DAC HAT, so I used an external USB DAC instead.

And boy did I have problems with the USB DAC. I would have Bit Perfect turned on, and it would turn off after a song was done playing. Then the DAC would stop outputting sound unless I went into the player settings and changed some setting. Then it would work again.

I don't think this is a problem with raudio1 itself. I think it's the underlying OS disconnecting and reconnecting the DAC.

2 days ago I got a Smartipi case, which let me put he HiFiBerry DAC2 HD back on the Pi, and all these problems have gone away.

But just a note that, if you're using a DAC with a Raspberry Pi and this wonderful software, you may want to get a DAC HAT instead of a USB DAC.

**Comment by R101G** [comment]
I have always used a USB DAC, and not had any problems. It might be the driver playing up, so it would be useful to know the make and model of DAC. It could also be that your setup has a unique conflict which does not affect anyone else. Does the USB DAC work without the touchscreen?

Are there any voltage warnings? A touchscreen and a USB DAC is going to draw quite a bit of current.

**Reply by apastuszak** [reply]
I'm using a JDS Labs OL-DAC with USB and SPDIF out. I did not get any voltage warning.

I believe it's a USB Audio Class 1 device, since it maxes out at 24/96.

**Reply by R101G** [reply]
I see it has its own PSU, so power should not be a problem. From what you say, it sounds as if the Pi is losing contact with the USB DAC, and is trying to connect to another output option.

**Reply by apastuszak** [reply]
That was my guess. Right now, the HiFiBerry DAC2 HD is working flawlessly, so I am not messing with anything.

**Reply by Roland-B** [reply]
Same here. Topping E30 USB DAC (has own USB power supply) and have no issues, but also have used many versions of DragonFly USB DACs  using power from the Pi with no issue either. My guess is power draw also, may be worth trying better power supply and quality USB/USB-C cable.

rAudio 1 • 20210628
5.10.44-1-ARCH aarch64
Raspberry Pi 4 Model B Rev 1.2
BCM2711 • 3.7GiB
4 @ 1.5GHz
